### PR TITLE
Use dedicated http client for publishing

### PR DESCRIPTION
**Pull request by avalonche**
## 📚 References

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [x] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

File: beaconclient/prod_beacon_instance.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ type ProdBeaconInstance struct {
 	// feature flags
 	ffUseV1PublishBlockEndpoint  bool
 	ffUseSSZEncodingPublishBlock bool
+
+	// http clients
+	publishingClient *http.Client
```
Comment (0x416e746f6e) [inline beaconclient/prod_beacon_instance.go:26]: **question:**

how is this client initialised?  as in, where do we create it and assign it to this field?

(tried to find it in the code, but couldn't.)

File: beaconclient/prod_beacon_instance.go [hunk 2]
```diff
@@ -179,7 +182,7 @@ func (c *ProdBeaconInstance) SyncStatus() (*SyncStatusPayloadData, error) {
 	uri := c.beaconURI + "/eth/v1/node/syncing"
 	timeout := 5 * time.Second
 	resp := new(SyncStatusPayload)
-	_, err := fetchBeacon(http.MethodGet, uri, nil, resp, &timeout, http.Header{}, false)
+	_, err := fetchBeacon(http.MethodGet, uri, nil, resp, &http.Client{Timeout: timeout}, http.Header{}, false)
```
Comment (0x416e746f6e) [inline beaconclient/prod_beacon_instance.go:185]: **question:**

why do we create a new client (and as a result a new TCP connection per each request) only for `/eth/v1/node/syncing` path?

File: beaconclient/prod_beacon_instance.go (hunk 1 above)
Comment (avalonche) [reply beaconclient/prod_beacon_instance.go:26]: this is 1 publishing http client per beacon node.

I can also have 1 publishing http client for all beacon nodes, let me know if that is better.

File: beaconclient/prod_beacon_instance.go (hunk 2 above)
Comment (avalonche) [reply beaconclient/prod_beacon_instance.go:185]: It's just to configure the timeout I think, will update to have 1 client in use for everything with a 5 second timeout (is that enough for the validator request?)

File: beaconclient/prod_beacon_instance.go (hunk 1 above)
Comment (avalonche) [reply beaconclient/prod_beacon_instance.go:27]: its initialised [here](https://github.com/flashbots/mev-boost-relay/pull/608/files#diff-77468d40b783b541b86c07535f04917be5a5b2436ea0278c6941317a4a46f270R37)?

**Review by metachris — APPROVED** [review]

**Pull request merged**